packages/devtools_app/lib/src/shared/common_widgets.dart

```diff
@@ -254,6 +248,166 @@ class ClearButton extends IconLabelButton {
         );
 }
 
+class EditableList extends StatefulWidget {
+  const EditableList({
+    required this.entries,
+    required this.label,
+    this.isRefreshing,
+    this.onEntryAdded,
+    this.onEntryRemoved,
+    this.onRefresh,
+  });
+
+  final ValueListenable<List<String>> entries;
+  final ValueListenable<bool>? isRefreshing;
+  final Function(String)? onEntryAdded;
+  final Function(String)? onEntryRemoved;
+  final Function()? onRefresh;
+  final String label;
+
+  @override
+  State<StatefulWidget> createState() => _EditableListState();
+}
+
+class _EditableListState extends State<EditableList> {
+  @override
+  void initState() {
+    super.initState();
+    textFieldController = TextEditingController();
+    widget.isRefreshing?.addListener(_stateRefresher);
```

Comment:
use addAutoDisposeListener (see AutoDisposeMixin). Then you don't have to manually dispose the listener. Another option, is to just use the DualValueListenableBuilder widget to listen to both `entries` and `isRefreshing` and then you don't have to manage adding listeners at all